packages/inter-protocol/test/psm/test-psm.js

```diff
@@ -305,6 +329,112 @@ test('limit', async t => {
   // t.throwsAsync(() => await E(seat1).getOfferResult());
 });
 
+/** @type {[kind: 'want' | 'give', give: number, want: number, result: number | false][]} */
+const trades = [
+  ['give', 200, 190, false],
+  ['want', 101, 100, 1],
+  ['give', 50, 50, false],
+  ['give', 51, 50, 1],
+];
+
+test('mix of trades: failures do not prevent later service', async t => {
+  const {
+    terms,
+    stable,
+    anchor,
+    feeMintAccess,
+    zoe,
+    installs: { centralSupply },
+  } = t.context;
+  const driver = await makePsmDriver(t);
+
+  const ist100 = await mintRunPayment(500n * 1_000_000n, {
+    centralSupply,
+    feeMintAccess,
+    zoe,
+  });
+
+  assert(anchor.issuer);
+  const anchorPurse = await E(anchor.issuer).makeEmptyPurse();
+  const stablePurse = await E(stable.issuer).makeEmptyPurse();
+  await E(stablePurse).deposit(ist100);
+
+  const scale6 = x => BigInt(Math.round(x * 1_000_000));
+
+  const wantStable = async (ix, give, want, result) => {
+    t.log('wantStable', ix, give, want, result);
+    const giveAnchor = AmountMath.make(anchor.brand, scale6(give));
+    const wantAmt = AmountMath.make(stable.brand, scale6(want));
+    const out = {};
+    const runPayouts = await driver.swapAnchorForStable(
+      giveAnchor,
+      wantAmt,
+      out,
+    );
+    if (result === false) {
+      await t.throwsAsync(out.offerResult);
+      return;
+    } else {
+      await out.offerResult;
+    }
+    t.is(await E(out.seat).numWantsSatisfied(), result);
+    if (result === 0) {
+      return;
+    }
+    const expectedRun = minusAnchorFee(giveAnchor, terms.anchorPerStable);
+    const actualRun = await E(stablePurse).deposit(await runPayouts.Out);
+    t.deepEqual(actualRun, expectedRun);
+  };
+
+  const giveStable = async (ix, give, want, result) => {
+    t.log('giveStable', ix, give, want, result);
+    const giveRun = AmountMath.make(stable.brand, scale6(give));
+    const runPayment = await E(stablePurse).withdraw(giveRun);
+    const wantAmt = AmountMath.make(anchor.brand, scale6(want));
+    const out = {};
+    const anchorPayouts = await driver.swapStableForAnchor(
+      giveRun,
+      runPayment,
+      wantAmt,
+      out,
+    );
+    if (result === false) {
+      await t.throwsAsync(out.offerResult);
+      return;
+    } else {
+      await out.offerResult;
+    }
+    t.is(await E(out.seat).numWantsSatisfied(), result);
+    if (result === 0) {
+      return;
+    }
+    const actualAnchor = await E(anchorPurse).deposit(await anchorPayouts.Out);
+    const expectedAnchor = AmountMath.make(
+      anchor.brand,
+      minusStableFee(giveRun).value,
+    );
+    t.deepEqual(actualAnchor, expectedAnchor);
+  };
+
+  let ix = 0;
+  for (const [kind, give, want, result] of trades) {
+    switch (kind) {
```

Comment:
this data driven test calls for [Ava macros](https://github.com/avajs/ava/blob/main/docs/01-writing-tests.md#reusing-test-logic-through-macros).  See https://github.com/Agoric/agoric-sdk/pull/5900

Reply:
Note that the items have to be used within one test. The whole point is to *not* make a new PSM between them. Do macros work that way?

Reply:
usually things that need be shared between tests are configured with `before` and `beforeEach`. but I won't block on this.

Reply:
it needs to be shared between these trades but no other tests

Reply:
then another file would work well. this `test` function is doing a lot of work to replicate some of the `before` and `macro` semantics.

Reply:
Let's punt on this for today.